experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -127,6 +126,7 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	public constructor(
 		private head: GraphCommit<TChange>,
 		public readonly changeFamily: ChangeFamily<TEditor, TChange>,
+		public repairDataStoreProvider: IRepairDataStoreProvider,
```

Comment:
So this is actually because of the `afterSummaryLoad` workaround that we have. Please let me know if you have any better ideas but basically, because the trunk's repair data store provider has a disjoint forest, when we load from a summary, it gets out of sync with main forest and we don't have enough information in the edit manager summary to create a new forest here. We also don't have a way to guarantee that the edit manager's load summary gets called after the forest's so we call this function manually after all the loading has been done to reclone.

Reply:
I don't really understand, but I'll take your word for it :D We can always revisit this later if things shuffle around enough to make another approach viable.